src/cdn_definitions/schema.yaml

```diff
@@ -11,7 +11,211 @@ $id: http://release-engineering.github.io/cdn-definitions/schema.json
+
+  arch_list:
+    type: array
+    items:
+      $ref: "#/definitions/arch"
+
+  release_version:
+    type: string
+    pattern: "^[0-9]+\\.[0-9]+$"
+
+  major_version:
+    type: string
+    pattern: "^[0-9]+$"
+
+  signing_key:
+    type: string
+    pattern: "^(37036783|A5787476|FD431D51|F21541EB|37017186|897DA07A|DB42A60E|\\[\\])$"
+
+  product_id:
+    type: string
+    pattern: "^[0-9]+$"
+
+  product_id_list:
+    type: array
+    items:
+      $ref: "#/definitions/product_id"
+
+  signing_key_list:
+    type: array
+    items:
+      $ref: "#/definitions/signing_key"
+
+  release_version_list:
+    type: array
+    items:
+      $ref: "#/definitions/release_version"
+
+  major_version_list:
+    type: array
+    items:
+      $ref: "#/definitions/major_version"
+
+  major_or_release_version:
+    type: string
+    pattern: "^[0-9]+(\\.[0-9]+)*$"
+
+  # various product definition mappings
+  arch_to_version_list_mapping:
+    type: object
+    propertyNames:
+      $ref: "#/definitions/arch"
+    patternProperties:
+      ".*":
+        $ref: "#/definitions/major_version_list"
+    additionalProperties: false
+
+  major_to_release_version_mapping:
+    type: object
+    propertyNames:
+      $ref: "#/definitions/major_version"
+    patternProperties:
+      ".*":
+        $ref: "#/definitions/release_version"
+    additionalProperties: false
+
+  release_stream_mapping:
+    type: object
+    properties:
+      aus:
+        $ref: "#/definitions/release_version_list"
+      eus:
+        $ref: "#/definitions/release_version_list"
+      tus:
+        $ref: "#/definitions/release_version_list"
+      e4s:
+        $ref: "#/definitions/release_version_list"
+    requiredProperties:
+    - aus
+    - eus
+    - tus
+    - e4s
+    additionalProperties: false
+
+  current_latest_rhel_versions:
+    type: object
+    propertyNames:
+      type: string
+      pattern: "^(current|latest)\\_rhel[0-9]+\\_(beta|dist)$"
+    patternProperties:
+      ".*":
+        $ref: "#/definitions/release_version"
+    additionalProperties: false
+
+  env_to_releasever_mapping:
+    type: object
+    properties:
+      qa:
+        $ref: "#/definitions/current_latest_rhel_versions"
+      stage:
+        $ref: "#/definitions/current_latest_rhel_versions"
+      prod:
+        $ref: "#/definitions/current_latest_rhel_versions"
+    required:
+    - qa
+    - stage
+    - prod
+    additionalProperties: false
+
+  product_cert_version_mapping:
+    type: object
+    propertyNames:
+      $ref: "#/definitions/major_version"
+    patternProperties:
+      ".*":
+        $ref: "#/definitions/release_version"
+    additionalProperties: false
+
+  # TPS stream configuration
+  rhel_ll_tps_stream_mapping:
+    type: object
+    propertyNames:
+      $ref: "#/definitions/release_version"
+    patternProperties:
+      ".*":
+        type: string
+        pattern: "^RHEL\\-\\%s\\-(LL|AUS)\\-\\%s$"
+    additionalProperties: false
+
+  uppercase_tps_variant_mappings:
```

Comment:
Could we perhaps not call this "uppercase"?

I'm looking at the related code now and I don't know why "upcase" is in the function name there, since it doesn't use uppercase values as inputs or outputs...

But anyway, even if the values were meant to be uppercase, I don't think that would need to go in the name unless there were some other non-uppercase mapping as well.